### DISCUSSION TITLE
Puppet HA question.

### DISCUSSION THREAD

**Discussion by vladsamoylik** [puppetlabs/community#49] [Q&A] [answered]
How can a puppet be deployed for complete fault tolerance?
What interests me:
- Duplication of certificate servers
- Duplication of servers with manifests
Use-case:
If one of the two servers is unavailable, clients must correctly obtain and sign certificates and manifests.

The documentation says that I can make a cluster, but there can only be one certificate authority. What makes the system not fault-tolerant.

**Comment by janfickler** [comment, marked as answer]
Hi,
for that case you mentioned you can use an NFS-Fileshare and mount in your system 

```
/etc/puppetlabs/puppetserver/ca
/opt/puppetlabs/puppet/cache/r10k
/etc/puppetlabs/code
```
depends on which OS you are.

then you can try to install your 2 puppetservers as PuppetCA and set them before a Loadbalancer and let your clients connect to them via Loadbalancer-URL.

**Comment by vladsamoylik** [comment]
Will try, thank you.
